supplier/models.py

```diff
@@ -51,7 +53,7 @@ def delete_supplier(supplier):
         raises:
         ObjectDoesNotExists error: if the supplier is not in DB.
         """
-        Supplier.objects.get(user_name=supplier.user_name).delete()
+        User.objects.get(username=supplier.user_name).delete()
```

Comment:
The method is called `delete_supplier`, but to me it looks like we're deleting a user now. No?

Reply:
The user field is cascade so since each supplier or client created with an account,if we delete an account so it will delete the specific sup or cli automatically and that is what we want

Reply:
I suggest explaining that in a comment.
```suggestion
        # The Supplier object will be deleted automatically once the User object it references is deleted.
        User.objects.get(username=supplier.user_name).delete()
```

Reply:
Same goes for the client
(I believe you could have just implemented a `delete_user` method on the `CustomUser` class, but I leave it for you to choose your preference)

Reply:
For more understanding i will left this as it is,comment has been added.
Fixed!